File: REST/src/exceptions/InvalidDataException.php
<?php

namespace Exceptions;

class InvalidDataException extends RestException {

	private $invalidField;


	/**
	 * Constructor
	 *
	 * @param	string     	$invalidField	invalid field name
	 */
	public function __construct(string $invalidField) {
		parent::__construct(self::BAD_REQUEST, "INVALID_DATA");

		$this->invalidField = $invalidField;
	}


	/**
	 * Get invalid field name
	 *
	 * @return	string		invalid field name
	 */
	public function getInvalidField() : string {
		return $this->invalidField;
	}


	/**
	 * Add exception specific data
	 *
	 * @param	array		$data		base level data
	 */
	protected function addErrorData(array &$data) {
		$data["field"] = $this->getInvalidField();
	}

}

File: common/src/main/java/it/unishare/common/exceptions/InvalidDataException.java
package it.unishare.common.exceptions;

public class InvalidDataException extends Exception {

    private static final long serialVersionUID = -7799565530950276949L;
    private String invalidField;


    /**
     * Constructor
     *
     * @param   invalidField        invalid field name
     */
    public InvalidDataException(String invalidField) {
        this.invalidField = invalidField;
    }


    /**
     * Get invalid field name
     *
     * @return  invalid field name
     */
    public String getInvalidField() {
        return invalidField;
    }

}
